packages/ui/src/components/cell/cell.native.tsx

```diff
@@ -1,113 +1,31 @@
-import { Switch } from 'react-native';
-
-import { ResponsiveValue, useTheme } from '@shopify/restyle';
-
-import { Box, ChevronRightIcon, IconProps, Text, Theme, TouchableOpacity } from '../../../native';
-
-type RegularCellVariant = 'active' | 'inactive' | 'critical';
-type RadioCellVariant = 'radio';
-type SwitchCellVariant = 'switch';
-type CellVariant = RegularCellVariant | RadioCellVariant | SwitchCellVariant;
-
-function getIconColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'radio':
-    case 'switch':
-      return 'ink.text-primary';
-    case 'inactive':
-      return 'ink.text-subdued';
-    case 'critical':
-      return 'red.action-primary-default';
-  }
-}
-
-function getIconBackgroundColor(
-  variant: CellVariant
-): ResponsiveValue<keyof Theme['colors'], Theme['breakpoints']> {
-  switch (variant) {
-    case 'active':
-    case 'radio':
-    case 'switch':
-      return 'ink.background-secondary';
-    case 'inactive':
-      return 'ink.background-secondary';
-    case 'critical':
-      return 'red.background-primary';
-  }
-}
-
-interface RadioCell {
-  variant: RadioCellVariant;
+import { TouchableOpacity } from '../button/touchable-opacity.native';
+import { ItemLayout } from '../item-layout/item-layout.native';
+
+interface CellWithoutAvatarProps {
+  actionIcon?: React.ReactNode;
+  captionLeft?: string;
+  captionRight?: string;
+  onPress?(): void;
+  titleLeft: string;
+  titleRight?: string;
```

Comment:
Thanks for refactoring @fbwoolf 

This makes me question whether we need "Cell" at all. It's just a wrapper around existing components, that feeds the props down. 

I don't think it adds any value versus composing this in every place we need it. 

The concept of _Cell_ could be a design-only concept that we simply implement on a case-by-case basis. Sometimes it's pressable, sometimes it isnt'. Sometimes has ItemLayout, sometimes has Flag only. There are too many variants to assume ahead of time, and I see no problem in using 2–3 extra lines of JSX for a Cell definition in mobile, if it provides us this flexibility and avoids us making assumptions.

@fabric-8 @mica000

Reply:
Yeah, it is a bit weird ...we also create our own `TouchableOpacity` so quite a few components. `ItemLayout` for mobile is interesting bc in our web version we always assume it needs to be a `Flag`, but that is not the case in mobile. So, maybe in mobile we just remove the `Flag` from `ItemLayout`, which is what I've done here, and compose them together where needed?

I'll remove my changes to `Cell`, and my use of it here. If it ends up working out better, I'll remove `Cell` in a different PR.